nerfstudio/model_components/ray_generators.py

```diff
@@ -52,7 +52,7 @@ def forward(self, ray_indices: TensorType["num_rays", 3]) -> RayBundle:
         camera_opt_to_camera = self.pose_optimizer(c)
 
         ray_bundle = self.cameras.generate_rays(
-            camera_indices=c,
+            camera_indices=c.unsqueeze(-1),
```

Comment:
Looked back at it, we do need this unsqueeze since the way we're generating rays (that allows for suuuuper easy index selection if we have variable dimension cameras) has the last dimension being the indices into the camera... so if we have 2 batch dimensions for our camera, and we want to generate a ray for cameras [0, 1] and [1,1], we pass in camera_indices=[[0,1], [1,1]]. Thus, if we are given a 1d tensor with all our camera indices, we need to add one extra dimension

Reply:
Gotcha, makes sense

Reply:
Gotha. We discussed offline. You can update the comments to explain the input in `generate_rays`.